tsl/src/data_node.c

```diff
@@ -574,10 +575,13 @@ connect_for_bootstrapping(const char *node_name, const char *const host, int32 p
 static void
 data_node_validate_extension_availability(TSConnection *conn)
 {
-	PGresult *res = remote_connection_execf(conn,
-											"SELECT default_version, installed_version FROM "
-											"pg_available_extensions WHERE name = %s",
-											quote_literal_cstr(EXTENSION_NAME));
+	bool compatible;
+
+	PGresult *res =
+		remote_connection_execf(conn,
+								"SELECT version FROM pg_available_extension_versions WHERE name = "
+								"%s AND version ~ '\\d+.\\d+.\\d+.*' ORDER BY version DESC",
```

Comment:
Not sure how Postgres stores versions here, for example is `2.0` will be always `2.0.0`?

Reply:
From what I can tell Postgres only stores string so we control the versioning. The order is not particularly important (code would work without ordering as well). I've only added it since most users would run the latest version so this would speed up the loop below.